src/reduced_order/nearest_neighbors.cpp

```diff
@@ -0,0 +1,85 @@
+#include "nearest_neighbors.h"
+
+namespace PHiLiP {
+namespace ProperOrthogonalDecomposition {
+
+NearestNeighbors::NearestNeighbors()
+        : snapshot_params()
+        , scaler()
+        , snapshots()
+{}
+
+void NearestNeighbors::updateSnapshots(const MatrixXd &snapshot_parameters, dealii::LinearAlgebra::distributed::Vector<double> snapshot){
+    snapshot_params = snapshot_parameters;
```

Comment:
I am copying to snapshot_param to keep a copy of the current snapshot parameters in the NearestNeighbors class since it's used in other functions.
I would prefer not to call `updateSnapshots()` in the constructor because this class is used throughout the pod_adaptive_sampling.cpp test, and every time a snapshot is collected it is added using the updateSnapshots() function. However, if it gets called in the constructor then the NearestNeighbors object needs to be created after the first snapshot is collected, rather than before collecting any snapshots as is done now. That would involve making an exception for the first snapshot collected (not to call updateSnapshots() twice) which would not be very elegant.